packages/table/components/Table.tsx

```diff
@@ -0,0 +1,172 @@
+import { AutoSizer, MultiGrid } from "react-virtualized";
+import { IColumnProps, Column } from "./Column";
+import memoizeOne from "memoize-one";
+import { coreColors as Colors } from "../../shared/styles/color";
+
+export interface ITableProps {
+  /**
+   * data is expecting an array of any values. These values should contain all the data represented in the table.
+   */
+  data: any[];
+  /**
+   * children can only be of type Column
+   */
+  children:
+    | Array<React.ReactElement<IColumnProps>>
+    | React.ReactElement<IColumnProps>;
+}
+
+const rowHeight = 35;
+export class Table<T> extends React.PureComponent<ITableProps, {}> {
+  public multiGridRef: { recomputeGridSize?: any } = {};
+
+  public getData = memoizeOne(
+    (data: T[]): Array<{}> | T[] => {
+      return [{}, ...data];
+    }
+  );
+
+  public getColumnSizes = memoizeOne(
+    (
+      children: Array<React.ReactElement<IColumnProps>>,
+      width: number
+    ): number[] => {
+      const totalColumns: number = children.length;
+      let remainingWidth: number = width;
+      return children.map((child, currentIndex) => {
+        const calculatedWidth = child.props.width({
+          width,
+          totalColumns,
+          currentIndex,
+          remainingWidth
+        });
+        remainingWidth -= calculatedWidth;
+        return calculatedWidth;
+      });
+    }
+  );
+
+  constructor(props) {
+    super(props);
+
+    this.getCell = this.getCell.bind(this);
+    this.setRef = this.setRef.bind(this);
+    this.updateGridSize = this.updateGridSize.bind(this);
+  }
+
+  public render() {
+    const mergedData = this.getData(this.props.data);
+
+    const columnCount = React.Children.count(this.props.children);
+
+    return (
+      <AutoSizer
+        defaultWidth={1024}
+        defaultHeight={768}
+        onResize={this.updateGridSize}
+      >
+        {({ width, height }) => {
+          const columnSizes = this.getColumnSizes(
+            React.Children.toArray(this.props.children) as Array<
+              React.ReactElement<IColumnProps>
+            >,
+            width
+          );
+
+          const getColumnSize = ({ index }) => columnSizes[index];
+          return (
+            <MultiGrid
+              ref={this.setRef}
+              fixedColumnCount={1}
+              fixedRowCount={1}
+              cellRenderer={this.getCell}
+              columnWidth={getColumnSize}
+              columnCount={columnCount}
+              enableFixedColumnScroll={true}
+              enableFixedRowScroll={true}
+              height={height}
+              rowHeight={rowHeight}
+              rowCount={mergedData.length}
+              width={width}
+              hideTopRightGridScrollbar={true}
+              hideBottomLeftGridScrollbar={true}
+            />
+          );
+        }}
+      </AutoSizer>
+    );
+  }
+
+  private getHeaderCell({
+    key,
+    style,
+    column
+  }: {
+    style: any;
+    key: any;
```

Comment:
Let's please avoid `any` esp for a new components we write from the scratch. It nullifies the whole point of using TS 🙏

Reply:
I'll do. Except if it is absolutely necessary, for example, data as this can basically be an array of any value. 

But in this case, I do agree not necessary.